### PR TITLE
stp: migrate to python@3.13

### PR DESCRIPTION
**Pull request by Moisan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/s/stp.rb [hunk 1]
```diff
@@ -40,12 +40,12 @@ class Stp < Formula
   depends_on "cryptominisat"
   depends_on "gmp"
   depends_on "minisat"
-  depends_on "python@3.12"
+  depends_on "python@3.13"
```
Comment (carlocab) [inline Formula/s/stp.rb:43]: I know we didn't have one previously, but would be good to add a minimal test for the Python bindings.

**Review by carlocab — APPROVED** [review]

File: Formula/s/stp.rb (hunk 1 above)
Comment (Moisan) [reply Formula/s/stp.rb:43]: Good idea, I added a test and there's something wrong going on with the Python bindings. It's failing on loading `library_path` which is in the installed in the `stp` `site-packages` folder. It seems that cannot be found by the `python3.13` executable.

Comment (Moisan) [reply Formula/s/stp.rb:43]: Importing `stp` from within the `site-packages` directory works so it seems to be a matter of getting the `PYTHON_PATH` correctly.

**Comment by Moisan** [comment]
Thanks a lot @daeho-ro !

**Review by p-linnane — APPROVED** [review]
Thanks @Moisan & @daeho-ro!

**Pull request merged**